src/syscheckd/CMakeLists.txt

```diff
@@ -131,5 +134,8 @@ if(UNIT_TEST)
 else()
   if(FSANITIZE)
     target_link_libraries(wazuh-syscheckd gcov asan ubsan)
+  elseif(CMAKE_BUILD_TYPE STREQUAL "Debug" AND NOT ${CMAKE_SYSTEM_NAME} MATCHES "Windows")
+    # Link against gcov when coverage is enabled in Debug mode
+    target_link_libraries(wazuh-syscheckd gcov)
```

Comment:
I don't understand why is this code necessary. `lcov` is already linked the line above.